--- s-os_lib.adb	2012-08-01 23:55:09.357901307 +0300
+++ s-os_lib-djgpp.adb	2012-06-23 13:36:27.000000000 +0300
@@ -57,7 +57,7 @@
 
    --  The following are used by Create_Temp_File
 
-   First_Temp_File_Name : constant String := "GNAT-TEMP-000000.TMP";
+   First_Temp_File_Name : constant String := "GA000000.TMP";
    --  Used to initialize Current_Temp_File_Name and Temp_File_Name_Last_Digit
 
    Current_Temp_File_Name : String := First_Temp_File_Name;
